docs/maven-tutorial.md

```diff
@@ -0,0 +1,218 @@
+	--claim-name 'nexus-pv' \
+	--claim-size '1G'
+```
+
+This removes the previous volume for the Deployment Config, and adds a claim
+for 1 gigabyte of storage mounted at `/sonatype-work`, which is where our
+dependencies will be stored.  Due to the change in config, the Nexus image will
+be redeployed automatically.
+
+Verify that Nexus is running by refreshing the Nexus page in your browser.
+
+## Modifying the Builder
+
+The next steps will be modifying our builder image to use this repository.
+The rest of the tutorial will be editing
+[this sample repository](https://github.com/bparees/openshift-jee-sample.git),
+but these changes should work for any project.
+
+Create a directory in your project called 'wildfly-builder'.  Inside this
+direcotry, create a file in your project called `settings.xml` with the
+following content:
+```xml
+<settings>
+  <mirrors>
+    <mirror>
+      <!--This sends everything else to /public -->
+      <id>nexus</id>
+      <mirrorOf>*</mirrorOf>
+      <url>http://nexus:8081/content/groups/public</url>
+    </mirror>
+  </mirrors>
+  <profiles>
+    <profile>
+      <id>nexus</id>
+      <!--Enable snapshots for the built in central repo to direct -->
+      <!--all requests to nexus via the mirror -->
+      <repositories>
+        <repository>
+          <id>central</id>
+          <url>http://central</url>
+          <releases><enabled>true</enabled></releases>
+          <snapshots><enabled>true</enabled></snapshots>
+        </repository>
+      </repositories>
+     <pluginRepositories>
+        <pluginRepository>
+          <id>central</id>
+          <url>http://central</url>
+          <releases><enabled>true</enabled></releases>
+          <snapshots><enabled>true</enabled></snapshots>
+        </pluginRepository>
+      </pluginRepositories>
+    </profile>
+  </profiles>
+  <activeProfiles>
+    <!--make the profile active all the time -->
+    <activeProfile>nexus</activeProfile>
+  </activeProfiles>
+</settings>
+```
+
+This is the crucial file for the effort.  We're telling Maven to use only the
+repository at nexus:8081, which is our proxy.  More information on DNS
+resolution in OpenShift can be found
+[here](https://docs.openshift.com/enterprise/3.1/architecture/additional_concepts/networking.html)
+
+Create a Dockerfile in the wildfly-builder directory with a `FROM` entry that
+points to the builder image you normally use.  Add a `COPY` entry pointing to
+the `settings.xml` you created earlier.
+
+```
+FROM docker.io/openshift/wildfly-100-centos7:latest
+
+COPY settings.xml /usr/share/maven/conf/settings.xml
+COPY settings.xml /opt/app-root/src/.m2/settings.xml
+```
+
+> NOTE: Maven's default global settings are in `/usr/share/maven/conf`, but some images
+> specify a settings.xml file when it is called.  For example, our WildFly builder
+> image uses `mvn -S /opt/app-root/src/.m2/settings.xml`, so we overwrite that
+> file with our own.  You may have to do the same.
```

Comment:
>We could overwrite the user settings, is there a way to get the $USER value in a pod from a Dockerfile?

You're using wildfly image as example, so I think that it will be enough to mention that in other cases $HOME dir can be different.

>Or perhaps we could append the content to an existing one.

No, you can't because XML has a structure. You can merge them, but it's hard, especially automatically.